#175636835 Setup dotenv Variables

Duncanwar opened the pull request:
#### What does this PR do?
It sets up environment variables for Barefoot nomad
#### Description of Task to be completed?
- Created `.env` file to hold all our environment variables
- Created `.env.example` file to show environment variables. You can change the text and add your suitable data
#### How should this be manually tested

run
- `git clone https://git.github.com/atlp-rwanda/tech-strikers-bn-backend`
- `cd tech-strikers-bn-backend`
- `git checkout ch-setup-dotenv-175636835`
- `npm install`
- create `.env` in root folder in this file create this variable
` API_VERSION = 'v1'`
- `npm run dev` and you will see in terminal v1
#### Any background context you want to provide?
All env variables are set up according to my local machine Postgres
#### What are the relevant pivotal tracker stories?
[#175636835](https://www.pivotaltracker.com/story/show/175636835)
#### Screenshots (if appropriate)
![Screenshot from 2020-11-11 12-59-48](https://user-images.githubusercontent.com/57761095/98811802-14bba400-242a-11eb-8f13-62bd36279009.png)
![Screenshot from 2020-11-11 12-59-42](https://user-images.githubusercontent.com/57761095/98811812-16856780-242a-11eb-88c8-32094c3abfff.png)
![Screenshot from 2020-11-11 12-59-31](https://user-images.githubusercontent.com/57761095/98811814-17b69480-242a-11eb-9557-416bcd65a590.png)

#### Questions:

MChristian00 reviewed (changes requested):
Everything looks fine.
Maybe you can change the structure of your PR Description on "**How should this be manually tested?**" as the points are not well structure

hervebutera reviewed (approved):
#### The PR comment is descriptive, but add `cd `  on how to change the directory to `tech-strikers-bn-backend`.  Otherwise the .env and .env.example don't have issues.

Duncanwar committed:
chore(dotenv):setup environment variables
- Create .env to hold all environment variables
- Create .env.sample to show all envirnoment variable without actual value
[Delivers #175636835]

kipropbrian commented on package.json:
```diff
@@ -24,14 +25,23 @@
     "underscore": "^1.9.1"
   },
   "devDependencies": {
+    "@babel/cli": "^7.12.1",
+    "@babel/core": "^7.12.3",
+    "@babel/node": "^7.12.6",
+    "@babel/preset-env": "^7.12.1",
```
These dependencies should not be in `devDepedencies` as they will not run in a production environment.

kipropbrian commented on package.json:
```diff
@@ -24,14 +25,23 @@
     "mongoose": "^5.2.2",
     "mongoose-unique-validator": "^2.0.1",
     "morgan": "^1.10.0",
+    "mysql2": "^2.2.5",
```
do we need this dependency?

kipropbrian commented on package.json:
```diff
@@ -24,14 +25,23 @@
     "passport-local": "^1.0.0",
+    "pg": "^8.5.0",
     "request": "^2.87.0",
+    "sequelize": "^6.3.5",
```
Please remove sequelize code. That should be the work of tytyne

Duncanwar commented:
Some dependency like those for mongoose are from the first version of the
application

On Mon, Nov 16, 2020 at 4:30 PM kipropbrian <notifications@github.com>
wrote:

> *@kipropbrian* commented on this pull request.
> ------------------------------
>
> In package.json
> <https://github.com/atlp-rwanda/tech-strikers-bn-backend/pull/4#discussion_r524309069>
> :
>
> >      "request": "^2.87.0",
> +    "sequelize": "^6.3.5",
>
> Please remove sequelize code. That should be the work of tytyne
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/atlp-rwanda/tech-strikers-bn-backend/pull/4#pullrequestreview-531372220>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ANYV2RY57BOYUKN4CVEJINLSQEZSHANCNFSM4TR54OXA>
> .
>

kipropbrian commented on src/config/config.js:
```diff
@@ -0,0 +1,46 @@
+require("dotenv").config();
+
+const {
+  DEV_DB_USERNAME,
+  DEV_DB_PASSWORD,
+  DEV_DB_NAME,
+  DEV_DB_HOSTNAME,
+  DEV_DB_PORT,
+  TEST_DB_USERNAME,
+  TEST_DB_PASSWORD,
+  TEST_DB_NAME,
+  TEST_DB_HOSTNAME,
+  TEST_DB_PORT,
+  PROD_DB_USERNAME,
+  PROD_DB_PASSWORD,
+  PROD_DB_NAME,
+  PROD_DB_HOSTNAME,
+  PROD_DB_PORT
+} = process.env;
```
I don't think we need this part. calling `config()` should load all of the variables. Otherwise every time something new is added we have to add it here.
